### PR TITLE
fix: remove reel related texture from memory when not used

### LINKED ISSUE #3829: fix: remove reel related texture from memory when not used

### PR DESCRIPTION
**Pull request by lorenzo-ranciaffi**
# Pull Request Description

Fixes #3829 

After memory profiling the E@ it was noticed that reels (both compressed and uncompressed versions) stayed in memory even if they were not used.

## What does this PR change?

This PR explicitly destroys/remove references (in the case of pooled objects) of every reel related texture when they are no longer used.

### Test Steps
1. Launch the E@
2. Verify that the reels work as before (in gallery, photo detail, users passports, navmpap places, etc)

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [x] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [x] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Code Review Standards](https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md) before submitting.

### CODE REVIEW

**Review by AnsisMalins — APPROVED** [review]

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs [hunk 1]
```diff
@@ -147,6 +147,8 @@ protected override void OnViewClose()
             HideDeleteModal();
 
             viewInstance.mainImageCanvasGroup.alpha = 0;
+            GameObject.Destroy(viewInstance!.mainImage.texture);
```
Comment (davidejensen) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:150]: small question here, what happens if you close the panel and force the destroy when it is still loading? don't you risk a nullref?

**Review by davidejensen — APPROVED** [review]

**Review by DafGreco — APPROVED** [review]
✔️  PR reviewed and approved by QA on both platforms following instructions playing both happy and un-happy path

Regressions for this ticket had been performed in order to verify that the normal flow is working as expected:

- ✅ Backpack and wearables in world
- ✅Emotes in world and in backpack
- ✅ Teleport with map/coordinates/Jump In
- ✅ Profile card
- ✅ Camera
- ✅ Skybox
- ✅Settings

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs (hunk 1 above)
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:150]: Fair question. Apparently it doesn't complain but it seems not a guarantee (may depend on Unity version). I'm adding a null check.
